packages/runtime/container-runtime/src/gc/gcConfigs.ts

```diff
@@ -63,21 +64,25 @@ export function generateGCConfigs(
 	 * 1. Whether running GC mark phase is allowed or not.
 	 * 2. Whether running GC sweep phase is allowed or not.
 	 * 3. Whether GC session expiry is enabled or not.
-	 * For existing containers, we get this information from the createParams.metadata blob of its summary.
+	 * For existing containers, we get this information from the metadata blob of its summary.
 	 */
 	if (createParams.existing) {
-		gcVersionInBaseSnapshot = getGCVersion(createParams.metadata);
-		// Existing documents which did not have createParams.metadata blob or had GC disabled have version as 0. For all
+		const metadata = createParams.metadata;
+		gcVersionInBaseSnapshot = getGCVersion(metadata);
+		// Existing documents which did not have metadata blob or had GC disabled have version as 0. For all
 		// other existing documents, GC is enabled.
 		gcEnabled = gcVersionInBaseSnapshot > 0;
-		sessionExpiryTimeoutMs = createParams.metadata?.sessionExpiryTimeoutMs;
-		sweepTimeoutMs =
-			createParams.metadata?.sweepTimeoutMs ?? computeSweepTimeout(sessionExpiryTimeoutMs); // Backfill old documents that didn't persist this
-		persistedGcFeatureMatrix = createParams.metadata?.gcFeatureMatrix;
+		sessionExpiryTimeoutMs = metadata?.sessionExpiryTimeoutMs;
+		const legacyPersistedSweepTimeoutMs = (metadata as IGCMetadata_Deprecated)?.sweepTimeoutMs;
+		tombstoneTimeoutMs =
+			metadata?.tombstoneTimeoutMs ??
+			legacyPersistedSweepTimeoutMs ?? // Backfill old documents that have sweepTimeoutMs instead of tombstoneTimeoutMs
+			computeTombstoneTimeout(sessionExpiryTimeoutMs); // Backfill old documents that didn't persist either value
```

Comment:
Here's the key backfill code. Other than this (and the addition of `interface IGCMetadata_Deprecated`, it's all just mechanical rename